storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdatesFactory.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.store;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.datastructures.blocks.SlotAndBlockRoot;
+import tech.pegasys.teku.datastructures.hashtree.HashTree;
+import tech.pegasys.teku.datastructures.state.BeaconState;
+import tech.pegasys.teku.datastructures.state.CheckpointAndBlock;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.storage.events.FinalizedChainData;
+import tech.pegasys.teku.storage.store.Store.Transaction;
+
+public class StoreTransactionUpdatesFactory {
```

Comment:
Don't you think that `Factory` is a bit misleading here? May be a `Builder` would be more appropriate?

Reply:
Yeah - I was unsure of the naming as well.  But I think "builder" implies that you can set various properties when externally it just looks like a factory:  `StoreTransactionUpdatesFactory.create(...)`

Reply:
Yeah I just used to the `Factory` pattern is when it's `create()` method is an _instance method_ so you could supply another factory implementation. Anyway since it's now package private there wouldn't be much confusion I think